app/controllers/api/base_controller.rb

```diff
@@ -281,8 +282,10 @@ def assign_lone_taxonomies
     end
 
     def add_version_header
-      response.headers["Foreman_version"] = SETTINGS[:version].full
-      response.headers["Foreman_api_version"] = api_version
+      response.headers['Foreman_api_version'] = SETTINGS[:api_version]
+      if Setting[:expose_version] && User.current.present?
```

Comment:
We should add a test for the enabled/disabled setting, and for the case without a logged-in user.